ext/sodium/libsodium.c

```diff
@@ -1662,20 +1606,24 @@ PHP_FUNCTION(sodium_crypto_pwhash)
 		sodium_remove_param_values_from_backtrace(EG(exception));
 		RETURN_THROWS();
 	}
-	if (hash_len <= 0 || hash_len >= 0xffffffff) {
-		zend_throw_exception(sodium_exception_ce, "hash length must be greater than 0", 0);
+	if (hash_len <= 0) {
+		zend_argument_error(sodium_exception_ce, 1, "to be greater than 0");
+		RETURN_THROWS();
+	}
+	if (hash_len >= 0xffffffff) {
+		zend_argument_error(sodium_exception_ce, 1, "to be less than the maximum allowed length");
```

Comment:
I do wonder if we should be including the maximum length in message like this. `to be less than 0xffffffff` or `to be less than the maximum length of 0xffffffff`. Too much information? Those messages are pretty exotic though... hard to run into them unless you really try ;)

Reply:
Personally, I'd piss off if I saw a maximum length of `0xffffffff` in an error message, to be honest :D Given, these errors are exotic, I wouldn't care to display the maximum.